src/ProcessorWrapper.ts

```diff
@@ -2,10 +2,35 @@ import type { ProcessorOptions, Track, TrackProcessor } from 'livekit-client';
 import { TrackTransformer } from './transformers';
 import { waitForTrackResolution } from './utils';
 
+export interface ProcessorWrapperOptions {
+  /**
+   * Maximum frame rate for fallback canvas.captureStream implementation
+   * Default: 30
+   */
+  maxFps?: number;
+}
+
 export default class ProcessorWrapper<TransformerOptions extends Record<string, unknown>>
   implements TrackProcessor<Track.Kind>
 {
   static get isSupported() {
+    // Check for primary implementation support
+    const hasStreamProcessor =
+      typeof MediaStreamTrackGenerator !== 'undefined' &&
+      typeof MediaStreamTrackProcessor !== 'undefined';
+
+    // Check for fallback implementation support
+    const hasFallbackSupport =
+      typeof HTMLCanvasElement !== 'undefined' &&
+      typeof VideoFrame !== 'undefined' &&
+      'captureStream' in HTMLCanvasElement.prototype;
+
+    // We can work if either implementation is available
+    return hasStreamProcessor || hasFallbackSupport;
+  }
+
+  // Method to determine if the modern APIs are supported
+  static get hasModernApiSupport() {
```

Comment:
can we document the apis clearly so it's easy for a customer to choose not to offer the processor at all (or add a warning) if they know it will degrade performance?

I think it should be docstringed inline and also covered in the readme

Reply:
for performance observability, see https://github.com/livekit/track-processors-js/pull/63